From 90e61e60e73608bfe7252c4c29cdb38a06d9640c Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Fri, 24 Jul 2020 17:39:20 +0800
Subject: [PATCH] arm64: configs: update rockchip_linux_defconfig

There are some configs needed to update for rockchip_linux_defconfig.
As below:

1) UART number increase to 6, e.g: px30...
2) Enable HW_RANDOM_ROCKCHIP for fast ramdom number init
3) Enable some the missing configs for rk805
4) Enable ARM_ROCKCHIP_BUS_DEVFREQ config
5) Enable USB_CONFIGFS_F_UVC for UVC

Signed-off-by: Caesar Wang <wxt@rock-chips.com>
Change-Id: I880a4c59e40ba7b79f6f68fc39fca55919314e7b
---
 arch/arm64/configs/rockchip_linux_defconfig | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/configs/rockchip_linux_defconfig b/arch/arm64/configs/rockchip_linux_defconfig
index 60788362b512..da31f5c8d48d 100644
--- a/arch/arm64/configs/rockchip_linux_defconfig
+++ b/arch/arm64/configs/rockchip_linux_defconfig
@@ -202,17 +202,19 @@ CONFIG_ROCKCHIP_REMOTECTL=y
 CONFIG_ROCKCHIP_REMOTECTL_PWM=y
 CONFIG_INPUT_MISC=y
 CONFIG_INPUT_UINPUT=y
+CONFIG_INPUT_RK805_PWRKEY=y
 # CONFIG_SERIO is not set
 CONFIG_VT_HW_CONSOLE_BINDING=y
 # CONFIG_LEGACY_PTYS is not set
 CONFIG_SERIAL_8250=y
 CONFIG_SERIAL_8250_CONSOLE=y
 # CONFIG_SERIAL_8250_PCI is not set
-CONFIG_SERIAL_8250_NR_UARTS=5
-CONFIG_SERIAL_8250_RUNTIME_UARTS=5
+CONFIG_SERIAL_8250_NR_UARTS=6
+CONFIG_SERIAL_8250_RUNTIME_UARTS=6
 CONFIG_SERIAL_8250_DW=y
 CONFIG_SERIAL_OF_PLATFORM=y
 CONFIG_HW_RANDOM=y
+CONFIG_HW_RANDOM_ROCKCHIP=y
 CONFIG_TCG_TPM=y
 CONFIG_TCG_TIS_I2C_INFINEON=y
 CONFIG_I2C_CHARDEV=y
@@ -223,6 +225,7 @@ CONFIG_SPI=y
 CONFIG_SPI_BITBANG=y
 CONFIG_SPI_ROCKCHIP=y
 CONFIG_SPI_SPIDEV=y
+CONFIG_PINCTRL_RK805=y
 CONFIG_DEBUG_GPIO=y
 CONFIG_GPIO_SYSFS=y
 CONFIG_GPIO_GENERIC_PLATFORM=y
@@ -390,6 +393,7 @@ CONFIG_USB_CONFIGFS_UEVENT=y
 CONFIG_USB_CONFIGFS_ACM=y
 CONFIG_USB_CONFIGFS_MASS_STORAGE=y
 CONFIG_USB_CONFIGFS_F_FS=y
+CONFIG_USB_CONFIGFS_F_UVC=y
 CONFIG_MMC=y
 CONFIG_MMC_BLOCK_MINORS=32
 CONFIG_MMC_TEST=y
@@ -429,6 +433,7 @@ CONFIG_PM_DEVFREQ=y
 CONFIG_DEVFREQ_GOV_PERFORMANCE=y
 CONFIG_DEVFREQ_GOV_POWERSAVE=y
 CONFIG_DEVFREQ_GOV_USERSPACE=y
+CONFIG_ARM_ROCKCHIP_BUS_DEVFREQ=y
 CONFIG_ARM_ROCKCHIP_DMC_DEVFREQ=y
 CONFIG_MEMORY=y
 CONFIG_IIO=y
-- 
2.35.3

